### PR TITLE
Docker support for ROSCon2023

### PR DESCRIPTION
**Pull request by spham-amzn**
## What does this PR do?
This adds docker support for the Warehouse (ROSCon2023Demo) that was shown at ROSCon 2023. 

## How was this PR tested?
Tested the steps that were described in the README.md that was added in the Docker subfolder

### CODE REVIEW

**Comment by spham-amzn** [comment]
Running the large simulation ROS client is currently failing to start the navigation of the otti robots after spawning them:
```
Staring robot otto_1
The passed service type is invalid
Staring robot otto_2
The passed service type is invalid
...
```

File: Docker/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+## Prerequisites
+
+* [Hardware requirements of o3de](https://www.o3de.org/docs/welcome-guide/requirements/)
+* Any Linux distribution that supports Docker and the Nvidia container toolkey (see below)
+* At least 60 GB of free disk space
+* Docker installed and configured
+  * **Note** It is recommended to have Docker installed correctly and in a secure manner so that the Docker commands in this guide do not require elevated priviledges (sudo) in order to run them. See [Docker Engine post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) for more details.
+* [NVidia container toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker)
+* [rocker](https://github.com/osrf/rocker)
```
Comment (pijaro) [inline Docker/README.md:11]: There are issues running O3DE with rocker, Ubuntu 22, and Nvidia drivers >525 (the Nvidia GPU is not recognized in the docker).

I proposed a fix for that, but it is still in PR: https://github.com/osrf/rocker/pull/277

Maybe it is worth mentioning here.

Comment (spham-amzn) [reply Docker/README.md:11]: Thats good to know, thanks

File: Docker/Dockerfile.O3DE [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+FROM ros:${ROS_VERSION}-ros-base-${UBUNTU_VERSION}
+
+ENV WORKSPACE=/data/workspace
+
+WORKDIR $WORKSPACE
+
+# O3DE Arguments
+ARG O3DE_REPO=https://github.com/o3de/o3de
+ARG O3DE_BRANCH=development
```
Comment (jhanca-robotecai) [inline Docker/Dockerfile.O3DE:18]: Shouldn't we keep the default configuration linked with the `main` releases? Or even the tag that was tested?

File: Docker/launch_ros.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+
+. /opt/ros/${ROS_DISTRO}/setup.sh
```
Comment (jhanca-robotecai) [inline Docker/launch_ros.sh:9]: I would either keep the shebang line pointing to `sh` or call `bash` scripts whenever possible.

**Review by jhanca-robotecai — APPROVED** [review]
Sorry for the late review, but I was sure the PR was already reviewed and merged.

Two minor comments, which are not really blockers for me.

File: Docker/Dockerfile.O3DE (hunk 2 above)
Comment (spham-amzn) [reply Docker/Dockerfile.O3DE:18]: I generally like to keep the same tag as the branch that the docker is in,  and in this case its the development branch of the demo. If this was merged into main, then the default branch tags here should do the same.

**Review by pijaro — APPROVED** [review]

**Pull request merged**